src/payment/strategies/checkoutcom/checkoutcom-payment-strategy.ts

```diff
@@ -0,0 +1,120 @@
+
+import { CheckoutcomToken } from './checkout';
+
+export default class CheckoutcomPaymentStrategy extends GooglePayPaymentStrategy {
+    TEST_URL: string = 'https://api.sandbox.checkout.com/tokens';
+    LIVE_URL: string = 'https://api.checkout.com/tokens';
+
+    constructor(
+        _store: CheckoutStore,
+        _checkoutActionCreator: CheckoutActionCreator,
+        _paymentMethodActionCreator: PaymentMethodActionCreator,
+        _paymentStrategyActionCreator: PaymentStrategyActionCreator,
+        _paymentActionCreator: PaymentActionCreator,
+        _orderActionCreator: OrderActionCreator,
+        _googlePayPaymentProcessor: GooglePayPaymentProcessor,
+        private _initializer: GooglePayCheckoutcomInitializer
+    ) {
+        super(_store,
+            _checkoutActionCreator,
+            _paymentMethodActionCreator,
+            _paymentStrategyActionCreator,
+            _paymentActionCreator,
+            _orderActionCreator,
+            _googlePayPaymentProcessor);
+    }
+
+    execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        if (!this.getGooglePayOptions()) {
+            throw new InvalidArgumentError('Unable to initialize payment because "options.googlepay" argument is not provided.');
+        }
+
+        const {
+            onError = () => {},
+            onPaymentSelect = () => {},
+        } = this.getGooglePayOptions();
+
+        return Promise.resolve(this.getPayment())
+            .then(payment => {
+                if (!payment.paymentData.nonce || !payment.paymentData.cardInformation) {
+
+                    return this.getGooglePayPaymentProcessor().displayWallet()
+                        .then(paymentData => this.getPaymentInstrumentSelected(paymentData))
+                        .then(() => onPaymentSelect())
+                        .then(() => this.getPayment())
+                        .catch(error => {
+                            if (error.statusCode !== 'CANCELED') {
+                                onError(error);
+                            }
+                        });
+                }
+
+            })
+            .then(() => {
+                    const payment = this.getPayment();
+                    let token: CheckoutcomToken | undefined;
+                    try {
+                        token = JSON.parse(payment.paymentData.nonce);
+                    } catch (err) {
+                        throw new InvalidArgumentError('Unable to parse response from Checkout.com.');
+                    }
+
+                    if ( token && token.protocolVersion ) {
+
+                        return this._requestCheckoutcomTokenize({
+                            type: 'googlepay',
+                            token_data: token,
+                        })
+                        .then(data => {
+                            payment.paymentData.nonce = data.token;
```

Comment:
A case when data does not contain a `token` is possible?  if so a validation would be nice 🍹

Reply:
It its, validation added